kura/org.eclipse.kura.net.admin/src/main/java/org/eclipse/kura/net/admin/monitor/DnsMonitorServiceImpl.java

```diff
@@ -217,25 +211,15 @@ protected void reconfigureDNSProxy(DnsServerConfigIP4 dnsServerConfigIP4) {
 
     @Override
     public void handleEvent(Event event) {
+        logger.info("Dns handle event");
         logger.debug("handleEvent - topic: {}", event.getTopic());
         String topic = event.getTopic();
 
         if (topic.equals(NetworkConfigurationChangeEvent.NETWORK_EVENT_CONFIG_CHANGE_TOPIC)) {
-            NetworkConfigurationChangeEvent netConfigChangedEvent = (NetworkConfigurationChangeEvent) event;
-            String[] propNames = netConfigChangedEvent.getPropertyNames();
-            if (propNames != null && propNames.length > 0) {
-                Map<String, Object> props = new HashMap<>();
-                for (String propName : propNames) {
-                    Object prop = netConfigChangedEvent.getProperty(propName);
-                    if (prop != null) {
-                        props.put(propName, prop);
-                    }
-                }
-                try {
-                    this.networkConfiguration = new NetworkConfiguration(props);
-                } catch (Exception e) {
-                    logger.warn(e.getMessage(), e);
-                }
+            try {
+                this.networkConfiguration = this.netConfigService.getNetworkConfiguration();
```

Comment:
As an enhancement we can consider moving this to a different thread to avoid performing blocking operations during an `EventAdmin` call.